File: sprint3/By languages/JS/lesson1code1.js
const fs = require('fs');
const path = require('path');

function find_file(folder, file_to_find) {
  const queue = [folder];
  while (queue.length > 0) {
    const item = queue.shift();
    if (fs.statSync(item).isDirectory()) {
      const files = fs.readdirSync(item);
      for (const file of files) {
        queue.push(path.join(item, file));
      }
    } else {
      if (path.basename(item) === file_to_find) {
        return item;
      }
    }
  }
  return null;
}

const folder = '/path/to/folder';
const file_to_find = 'file.txt';
const result = find_file(folder, file_to_find);
if (result === null) {
  console.log('File not found');
} else {
  console.log(`File found at ${result}`);
}

File: sprint3/By languages/C#/lesson1code1.cs
using System;
using System.Collections.Generic;
using System.IO;

class MainClass {
    public static string find_file(string folder, string file_to_find) {
        Queue<string> queue = new Queue<string>();
        queue.Enqueue(folder);
        while (queue.Count > 0) {
            string item = queue.Dequeue();
            if (Directory.Exists(item)) {
                string[] files = Directory.GetFiles(item);
                string[] subdirs = Directory.GetDirectories(item);
                foreach (string file in files) {
                    queue.Enqueue(file);
                }
                foreach (string subdir in subdirs) {
                    queue.Enqueue(subdir);
                }
            } else {
                if (Path.GetFileName(item) == file_to_find) {
                    return item;
                }
            }
        }
        return null;
    }

    public static void Main() {
        string folder = "/path/to/folder";
        string file_to_find = "file.txt";
        string result = find_file(folder, file_to_find);
        if (result == null) {
            Console.WriteLine("File not found");
        } else {
            Console.WriteLine("File found at {0}", result);
        }
    }
}
